components/apps.js

```diff
@@ -649,33 +647,152 @@ SteamUser.prototype.getOwnedDepots = function(excludeSharedLicenses) {
+		return this._returnPackages(packages);
+	}
+	// If filter options is a boolean, we asssume it's excludeSharedLicenses
+	if (typeof filter === 'boolean') {
+		let excludeSharedLicenses = filter;
+		filter = {excludeShared: excludeSharedLicenses};
+	}
+
+	// New behavior from this point on
+	// Handle filter options
+	if (typeof filter === 'object') {
+		const defaults = {
+			excludeFree: false, // By default, include free licenses (FreeOnDemand, NoCost, etc.)
+			excludeShared: false, // By default, include shared licenses
+			excludeExpiring: false // By default, include licenses that are going to expire in the future (free weekends)
+		};
+
+		if (typeof this.options.ownershipFilter === 'object') {
+			filter = Object.assign(defaults, this.options.ownershipFilter, filter);
+		} else {
+			filter = Object.assign(defaults, filter);
+		}
+
+		// No PICS cache needed: {F=0,S=1,E=0}, {F=0,S=0,E=0}
+		if (!filter.excludeFree && !filter.excludeExpiring) {
+			if (filter.exludeShared) {
+				packages = packages.filter(license => license.owner_id == this.steamID.accountid);
+			}
+			return this._returnPackages(packages);
+		}
+	}
+
+	// From this point, we need PICS cache to be enabled
+	this._ensurePicsCache();
+
+	// Determine the filter function
+	let packageFilter = null;
+	if (typeof filter === 'function') {
+		packageFilter = filter;
+	} else if (typeof filter === 'object') {
+		// Possible options: {F=1,S=1,E=1}, {F=1,S=0,E=1}, {F=0,S=1,E=1}, {F=0,S=0,E=1}, {F=1,S=1,E=0}, {F=1,S=0,E=0}
+		packageFilter = (license) => {
+
+			// If expired, filter it out, regardless of the filter options
+			// Assumes: license expired flag <=> package expirytime passed
+			if (license.flags & SteamUser.ELicenseFlags.Expired) {
+				return false;
+			}
+
+			// If exclude shared licenses
+			if (filter.excludeShared && license.owner_id !== this.steamID.accountid) {
+				return false;
+			}
+
+			let subid = license.package_id;
+			if (!this.picsCache.packages[subid] || !this.picsCache.packages[subid].packageinfo) {
+				this._warn(`Failed to filter package ${subid} (no PICS cache info available)`);
+				return false;
+			}
+
+			let pkg = this.picsCache.packages[subid].packageinfo;
+			let owned = true;
+
+			// If exclude all free (sub 0 is covered by NoCost)
+			if (filter.excludeFree) {
+				owned = owned
+						&& pkg.billingtype !== SteamUser.EBillingType.NoCost
+						&& pkg.billingtype !== SteamUser.EBillingType.GuestPass
+						&& pkg.billingtype !== SteamUser.EBillingType.FreeOnDemand
+						&& pkg.billingtype !== SteamUser.EBillingType.FreeCommercialLicense;
+			}
+
+			// If not temporary (free promotions are yours to keep permanently)
+			if (!pkg.extended || !pkg.extended.expirytime || pkg.extended.freepromotion) {
+				return owned;
+			}
+
+			// If exclude all expiring licenses
+			if (filter.excludeExpiring) {
+				return false; // return false, since this license is temporary (but not expired)
+			}
+
+			return owned;
```

Comment:
Can we simplify it a little?
```js
//let owned = true;

if (filter.excludeFree) {
	switch (pkg.billingtype) {
		case SteamUser.EBillingType.NoCost:
		case SteamUser.EBillingType.GuestPass:
		case SteamUser.EBillingType.FreeOnDemand:
		case SteamUser.EBillingType.FreeCommercialLicense:
			return false;
	}
}

// exclude temporary licenses, but keep free promotions?
if (filter.excludeExpiring && pkg.extended && pkg.extended.expirytime && !pkg.extended.freepromotion) {
	return false;
}

return true;
```

Reply:
And important question: should we consider `GuestPass` as "Free" or as "temporary"? 🤔 🤔 🤔

Reply:
Maybe both?
Btw, you can use ` ```suggestion  ... ``` ` to suggest code changes. Pretty neat feature.